src/pages/iou/IOUDetailsModal.js

```diff
@@ -146,47 +147,54 @@ class IOUDetailsModal extends Component {
 
     render() {
         const sessionEmail = lodashGet(this.props.session, 'email', null);
-        const reportIsLoading = _.isUndefined(this.props.iouReport);
+        const reportIsLoading = this.props.iou.loading;
```

Comment:
@Julesssss I think this makes sense, right? This was the one part I was unsure about - the difference between `iou` and `iouReport` and if they're linked enough anyway that the `loading` property from `iou` would suffice.

Reply:
I try to log the `this.props.iou.loading` inside the render function, but it returns `false` on both accounts (the account that has the IOU data and the account which doesn't have the IOU data).

Reply:
So what are we saying? This prop is just useless anyway?

Reply:
Could be, but `this.props.iou.loading` is also been used for disabling the settlement button. I will take a look more on the `props.iou`.

Reply:
Hmm, yeah that one might be deprecated actually. it's probably better to check for the report itself in Onyx

Reply:
> Hmm, yeah that one might be deprecated actually. it's probably better to check for the report itself in Onyx

What do we mean by this? `_.isUndefined(this.props.iouReport)`? That won't work for this situation because if we don't have access to the report it won't be in Onyx.

Reply:
> Alternatively, we can create single Onyx data to keep the loading state for the OpenPaymentDetailsPage API similar to what we did with the OpenApp API and ONYXKEYS.IS_LOADING_REPORT_DATA.

I think I like this - what do you think @Julesssss ?

Reply:
Yeah, I think that is pretty straight forward. Let's do it.

Reply:
Actually, on second look, I think we _don't_ need this since we already load all iouReports when opening the app, and if one is ever created for you it is pushed to you via Pusher. So there would never be a state where we would need to be "loading" the IOU report data? Right?

Reply:
Ah nevermind no, this would be something that needs to wrap the `OpenPaymentDetailsPage` command.

Reply:
Ah okay - 

> Could be, but this.props.iou.loading is also been used for disabling the settlement button. I will take a look more on the props.iou.

It's not useful anymore since we optimistically pay the IOU, so the settlement button's loading state is outdated. So, what I'll do is repurpose that `loading` property for loading the iou report itself.